build/generate/dtsFile.go

```diff
@@ -20,9 +24,50 @@ func generateDTSFile(funcs string) error {
 		if err != nil {
 			return err
 		}
+		// Find all instances of `/** @dnscontrol-auto-doc-comment <path> */`
```

Comment:
> Is the intention to use `@dnscontrol-auto-doc-comment` on all (or most) functions? (eventually)

It’s a bit of a balancing act — most of the JS APIs have pretty basic type signatures (which is great for understandability/usability!) and no manual work beyond writing the docs page with the right metadata is needed. But for the few exceptions listed above it would be nice to use the automation. For now, only `FETCH` has its own dedicated doc page to copy from and a complex enough type to warrant a custom declaration. (the CLI variables doc page doesn’t work very well as a doc comment so I just link to it)

Reply:
Sounds good! Thanks for putting up with all my naieve questions!